lib/internal/modules/esm/loader.js

```diff
@@ -138,14 +147,28 @@ class DefaultModuleLoader {
    * @returns {ModuleJob} The (possibly pending) module job
    */
   getModuleJob(specifier, parentURL, importAssertions) {
-    const resolveResult = this.resolve(specifier, parentURL, importAssertions);
+    const jobPromise = this.#getModuleJob(specifier, parentURL, importAssertions);
+    return {
+      run() {
+        return PromisePrototypeThen(jobPromise, (job) => job.run());
+      },
+      get modulePromise() {
+        return PromisePrototypeThen(jobPromise, (job) => job.modulePromise);
+      },
+      get linked() {
+        return PromisePrototypeThen(jobPromise, (job) => job.linked);
+      },
+    };
+  }
+
+  async #getModuleJob(specifier, parentURL, importAssertions) {
+    const resolveResult = await this.resolve(specifier, parentURL, importAssertions);
```

Comment:
I know it was the way you had it in the previous version, but `getModuleJob` and `#getModuleJob` feel like they should be just one method.

Why was this part of `CustomizedModuleLoader` previously but now all this logic needs to be in the default loader?

Reply:
> I know it was the way you had it in the previous version, but getModuleJob and #getModuleJob feel like they should be just one method.

There isn't really a good reason other than I was just following precedent; the previous customized module loaders used this approach so I just followed suit in this implementation. More than happy to combine!

Reply:
> Why was this part of CustomizedModuleLoader previously but now all this logic needs to be in the default loader?

a) Delegates can return promises, so `getModuleJob` has to be able to handle async results now.
b) `resolve` is now permanently async, so `getModuleJob` has to be able to handle async results now.

Reply:
Actually this made me realize I can simplify things quite dramatically. Changes coming in hot 🔥